modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerHeaderFragment.kt

```diff
@@ -437,7 +439,8 @@ class PlayerHeaderFragment : BaseFragment(), PlayerClickListener {
 
             val imageBuilder: ImageRequest.Builder.() -> Unit = {
                 error(IR.drawable.defaultartwork_dark)
-                size(Size.ORIGINAL)
+                scale(Scale.FIT)
+                size(CoilManager.getMaxImageSize(imageView))
```

Comment:
> it most likely loads bitmaps larger than needed because it is unlikely that we have ImageViews larger than 2500px.

I guess `Size.ORIGINAL` was initially used to display a high-resolution image on a full-screen player without any scaling, similar to how iOS utilizes a 3x resolution that exceeds the actual size of the image. For this reason, I allowed a maximum image size that could exceed the imageview, yet could be loaded comfortably.

> From my testing we can safely remove call to change the size. Can you check it as well and if everything works remove it?

I thought about large high-resolution tablets while restricting the size, but being larger devices, they should allow larger images. I tested on a few today: Pixel C (10.95" 2560x1600) and a custom Samsung tablet (14" 1848 x 2960), and they were able to load these images fine after scaling. To my eyes, images were not pixelated and still look good, so I've removed this size restriction in https://github.com/Automattic/pocket-casts-android/pull/1665/commits/530df4af8b7d8f7796ac6a920cb340896ac8c2d2.

Reply:
> For this reason, I allowed a maximum image size that could exceed the imageview, yet could be loaded comfortably.

I might be confused by something but wouldn't this code make the image *at least* 2500px? If we wanted to have it not exceed some size I think `min()` should be used instead of `max()`.

```kotlin
fun getMaxImageSize(imageView: ImageView) = Size(
    max(imageView.width, MAX_IMAGE_SIZE_IN_PX),
    max(imageView.height, MAX_IMAGE_SIZE_IN_PX)
)
```

Reply:
> wouldn't this code make the image at least 2500px

`loadEpisodeArtwork(..)` is currently just called for the Player screen with a big image view but so far I haven't seen exceed more than 2500px. The intention for this workaround was to keep the image size big enough (even for smaller screens) that could load comfortably yet not cause an OOM error. Nevertheless, we removed this part so I think we're good.